### PR TITLE
Add MODULE definition for x-pack/dockerlogbeat Go Integration tests

### PR DESCRIPTION
**Pull request by oakrizan**
## Proposed commit message
In scope of https://github.com/elastic/ingest-dev/issues/3254 - added MODULE env var definition for Go Integration tests to match same behavior as on Jenkins.
Jenkins vs. Buildkite comparison: https://gist.github.com/oakrizan/bf2aa5f1cd1ab01daa1184f471a99abc
BK build: https://buildkite.com/elastic/beats-xpack-dockerlogbeat/builds?branch=oakrizan%3Axpack-dockerlogbeat-fixes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml [hunk 1]
```diff
@@ -61,13 +68,13 @@ steps:
     # this allows building DRA artifacts even if there is flakiness in mandatory tests
     if: build.env("BUILDKITE_PULL_REQUEST") != "false"
     depends_on:
-      - "xpack-dockerlogbeat-mandatory-tests"
+      - "x-pack-dockerlogbeat-mandatory-tests"
 
-  - group: "Xpack/Dockerlogbeat Packaging"
-    key: "xpack-dockerlogbeat-packaging"
+  - group: "x-pack/Dockerlogbeat Packaging"
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml:73]:
```suggestion
  - group: "x-pack/dockerlogbeat Packaging"
```

File: .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml [hunk 2]
```diff
@@ -22,10 +22,10 @@ env:
   IMAGE_WIN_2022: "family/platform-ingest-beats-windows-2022"
 
 steps:
-  - group: "Xpack/Dockerlogbeat Mandatory Tests"
-    key: "xpack-dockerlogbeat-mandatory-tests"
+  - group: "x-pack/Dockerlogbeat Mandatory Tests"
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml:25]:
```suggestion
  - group: "x-pack/dockerlogbeat Mandatory Tests"
```

File: .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml [hunk 3]
```diff
@@ -22,10 +22,10 @@ env:
...
+    key: "x-pack-dockerlogbeat-mandatory-tests"
     steps:
-      - label: ":ubuntu: Xpack/Dockerlogbeat Ubuntu Unit Tests"
+      - label: ":ubuntu: x-pack/Dockerlogbeat Ubuntu Unit Tests"
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml:28]:
```suggestion
      - label: ":ubuntu: x-pack/dockerlogbeat Ubuntu Unit Tests"
```

File: .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml [hunk 4]
```diff
@@ -39,9 +39,16 @@ steps:
           - github_commit_status:
               context: "x-pack/dockerlogbeat: Ubuntu Unit Tests"
 
-      - label: ":ubuntu: Xpack/Dockerlogbeat Go Integration Tests"
+      - label: ":ubuntu: x-pack/Dockerlogbeat Go Integration Tests"
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml:42]:
```suggestion
      - label: ":ubuntu: x-pack/dockerlogbeat Go Integration Tests"
```

File: .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml [hunk 5]
```diff
@@ -61,13 +68,13 @@ steps:
...
+      - "x-pack-dockerlogbeat-mandatory-tests"
 
-  - group: "Xpack/Dockerlogbeat Packaging"
-    key: "xpack-dockerlogbeat-packaging"
+  - group: "x-pack/Dockerlogbeat Packaging"
+    key: "x-pack-dockerlogbeat-packaging"
 
     steps:
-      - label: ":ubuntu: Xpack/Dockerlogbeat Packaging Linux X86"
+      - label: ":ubuntu: x-pack/Dockerlogbeat Packaging Linux X86"
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml:77]:
```suggestion
      - label: ":ubuntu: x-pack/dockerlogbeat Packaging Linux X86"
```

File: .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml [hunk 6]
```diff
@@ -77,8 +84,11 @@ steps:
           provider: gcp
           image: "${IMAGE_UBUNTU_X86_64}"
           machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
+        notify:
+          - github_commit_status:
+              context: "x-pack/dockerlogbeat: Packaging Linux X86"
 
-      - label: ":linux: Xpack/Dockerlogbeat Packaging Linux ARM"
+      - label: ":linux: x-pack/Dockerlogbeat Packaging Linux ARM"
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml:91]:
```suggestion
      - label: ":linux: x-pack/dockerlogbeat Packaging Linux ARM"
```

**Review by dliappis — COMMENTED** [review]
Basically LGTM; I've left a few comments to lowercase `Dockerlogbeat` reflecting the name (and helping GH keep the status checks properly sorted).

File: .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml (hunk 1 above)
Comment (oakrizan) [reply .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml]: Updated

File: .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml [hunk 7]
```diff
@@ -59,39 +69,49 @@ steps:
...
+      - "x-pack-dockerlogbeat-mandatory-tests"
 
-  - group: "Xpack/Dockerlogbeat Packaging"
-    key: "xpack-dockerlogbeat-packaging"
+  - group: "x-pack/dockerlogbeat Packaging"
+    key: "x-pack-dockerlogbeat-packaging"
 
     steps:
-      - label: ":ubuntu: Xpack/Dockerlogbeat Packaging Linux X86"
+      - label: ":ubuntu: x-pack/dockerlogbeat Packaging Linux x86_64"
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml:86]:
```suggestion
      - label: ":ubuntu: x-pack/dockerlogbeat Packaging Linux"
```

File: .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml [hunk 8]
```diff
@@ -59,39 +69,49 @@ steps:
...
+        command: |
+          cd x-pack/dockerlogbeat
+          mage package
         agents:
           provider: gcp
           image: "${IMAGE_UBUNTU_X86_64}"
           machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
+        notify:
+          - github_commit_status:
+              context: "x-pack/dockerlogbeat: Packaging Linux x86_64"
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.dockerlogbeat.yml:100]:
```suggestion
              context: "x-pack/dockerlogbeat: Packaging Linux"
```

**Review by dliappis — APPROVED** [review]
LGTM

left a small ask to change the name of the main packaging step to "Packaging" because it is packaging everything apart from Linux ARM.

**Review by dliappis — APPROVED** [review]
LGTM

**Pull request merged**